File: app/src/main/java/com/example/kotlin/first/generic/test8/java/presenter/TestMvpActivityPresenter.java
package com.example.kotlin.first.generic.test8.java.presenter;

import com.example.kotlin.first.generic.test8.java.view.ITestMvpActivityView;

/**
 * MVP公共Presenter
 */
public class TestMvpActivityPresenter extends BaseMvpPresenter<ITestMvpActivityView> {

    public TestMvpActivityPresenter(ITestMvpActivityView view) {
        super(view);
    }

    public String getTestString() {
        return "TestMvpPresenter";
    }

    public void onTip() {
        getView().test();
    }

}

File: app/src/main/java/com/example/kotlin/first/generic/test8/kotlin/presenter/TestMvpActivityPresenter.kt
package com.example.kotlin.first.generic.test8.kotlin.presenter

import com.example.kotlin.first.generic.test8.kotlin.view.ITestMvpActivityView
import com.example.kotlin.first.generic.test8.kotlin.view.TestMvpActivity

class TestMvpActivityPresenter(view: TestMvpActivity) : BaseMvpPresenter<ITestMvpActivityView?>(view) {

    val testString: String
        get() = "TestMvpPresenter"

    fun onTip() {
        view?.test()
    }
}
